### ISSUE TITLE
User cant join meeting after compliance policy

### ISSUE THREAD

**Issue by aarushi-mittal** [microsoftgraph/microsoft-graph-comms-samples#187]
Hello ,

I have created a compliance policy for my user and now it is not able to get into a meeting.

The teams interface says -> connecting -> then sorry cant connect you

Kindly help

**Comment by aarushi-mittal** [comment]
@ksikorsk @mathowar

**Comment by mathowar** [comment]
Hello, this is most likely due to an error in your misconfiguration in setting up your cloud service. This would cause both the call to not connect and the logs to not show up. It's hard to say what exactly went wrong just from this. Can you please list all of the steps that you took?

**Comment by aarushi-mittal** [comment]
Hi @mathowar ,
I am using the following configurations from a successful attempt on AudioVideoPlaybackBot:
1. cloud service 
2. SSL cert and DNS
3. App registration and bot channel

I have a hitch that I maybe doing something wrong in the compliance policy construction, because as soon as I attach a policy to a user it starts to disconnect.

steps for compliance policy:

1. create application instance
`New-CsOnlineApplicationInstance -UserPrincipalName <upn@contoso.com> -DisplayName <displayName> -ApplicationId <your_botappId>`
here i have created a new name and my tenant is xyz. so teh upn = notname@xyz

2. grant policy to a user
`Grant-CsTeamsComplianceRecordingPolicy -Identity <userUnderPolicy@contoso.com> -PolicyName <policyIdentity>`
here the user is an existing user on this tenant. 
Now I am confused here also.
My subscription has been detached from a new o365 ID and attached to my corporate email id .
This email id(corporate, where I am tenant admin) was being used  in AudioVideoPlayback bot.

I was not able to create a policy using this id as I did not have perimissions to the tenant.
So I used another user that had similar priviledges on this subscription.

In short , what is needed on the user account permission side to create a policy successfully.

I thought something was wrong on tenant permission side so I tried with entirely different account. But lync was not able to discover the domain there.

Let me know in case you need more details to get a clearer idea 

regards,
Aarushi

**Comment by mathowar** [comment]
Hi Aarushi, I have a couple comments.

1. The fact that you are unable to get logs has nothing to do with the policy being assigned. This seems to suggest to me that there is still something wrong with the cloud service. What do you see when you try to GET https:<dns>.cloudapp.net/logs?

2. Can you please post all of the output when creating the policy? It is still unclear to me whether those were successful. I am not an expert on the cmdlets, but it is my understanding that you just need to be the tenant admin. 

3. It takes some time after creating a user for lync to be able to discover the account, so maybe you could try again?

Thanks,
Matt

**Comment by aarushi-mittal** [comment]
> Hi Aarushi, I have a couple comments.
> 
> 1. The fact that you are unable to get logs has nothing to do with the policy being assigned. This seems to suggest to me that there is still something wrong with the cloud service. What do you see when you try to GET https:.cloudapp.net/logs?

I do GET https://<custom-domain>.com/logs
and there is a success(from Postman its a 200 ok response) on the browser too but I get a blank screen.
there is absolutely nothing on it

> 2. Can you please post all of the output when creating the policy? It is still unclear to me whether those were successful. I am not an expert on the cmdlets, but it is my understanding that you just need to be the tenant admin.
Adding it in the next response
> 3. It takes some time after creating a user for lync to be able to discover the account, so maybe you could try again?
I tried it multiple times.
So here's the thing, the azure ad domain for that account had been corrected from xyz.onmicrosoft.com -> to -> xyzcom

Now lync doesnt get domain "xyz.com" and for "xyz.onmicrosoft.com" there is no azure ad 
PFA images for this in next response
> 
> Thanks,
> Matt

**Comment by aarushi-mittal** [comment]
Hello Matt,

PFB powershell output for point no 2 .
[compliance policy creation steps.txt](https://github.com/microsoftgraph/microsoft-graph-comms-samples/files/4314752/compliance.policy.creation.steps.txt)

PFB images for the point no 3 .
<img width="742" alt="failed lync discover" src="https://user-images.githubusercontent.com/28774537/76357557-f1da6c80-633d-11ea-8a07-112658f16caf.png">

Also :
1. what is the way to check if its a tenant admin (if the user has been added to a subscription, can they be tenant admin?)
2. what happens to the bot application user that gets created?

**Comment by doucheface1** [comment]
did you resolve this im having same issue?

**Comment by simonso1224** [comment]
I am at this stage now having the same issue. I checked that I am able to get to https://bot.mydomain.com/logs and see something useful after I dumped some garbage into https://bot.mydomain.com/api/calling, so I believe the webhook is working. 

What did you guys do/use to troubleshoot this issue? Thanks a lot.
